### PR TITLE
Private presale custom rates

### PR DESCRIPTION
**Pull request by AugustoL**

### CODE REVIEW

File: contracts/LifCrowdsale.sol [hunk 1]
```diff
@@ -238,13 +231,17 @@ contract LifCrowdsale is Ownable, Pausable {
 
      @param beneficiary Address to which Lif will be sent
      @param weiSent Amount of wei contributed
+     @param rate of tokens per ether at the moment of the contribution
```
Comment (nberger) [inline contracts/LifCrowdsale.sol]: I'd change it to:

`@param rate Lif per ether rate at the moment of the contribution`

(changed tokens to Lif, and also note that rate is the param name, and the rest is the description of the param, it's not supposed to be a regular phrase as a whole but first the param, then the description)

Comment (AugustoL) [reply contracts/LifCrowdsale.sol]: :+1:

File: contracts/LifCrowdsale.sol [hunk 2]
```diff
@@ -238,13 +231,17 @@ contract LifCrowdsale is Ownable, Pausable {
...
    */
-  function addPrivatePresaleTokens(address beneficiary, uint256 weiSent) onlyOwner {
+  function addPrivatePresaleTokens(
+    address beneficiary, uint256 weiSent, uint256 rate
+  ) onlyOwner {
     require(block.timestamp < startTimestamp);
     require(beneficiary != address(0));
     require(weiSent > 0);
+    require(rate > rate1);
```
Comment (nberger) [inline contracts/LifCrowdsale.sol]: I'd add a comment explaining this. It'd be something like "validate that rate is higher than TGE rate"

File: contracts/LifCrowdsale.sol [hunk 3]
```diff
@@ -238,13 +231,17 @@ contract LifCrowdsale is Ownable, Pausable {
...
 
-    uint256 tokens = weiSent.mul(privatePresaleRate);
+    uint256 tokens = weiSent.mul(rate);
```
Comment (nberger) [inline contracts/LifCrowdsale.sol]: Would adding an event here give an easier way to see the details of the presale contributions? Should we add one?

File: contracts/LifCrowdsale.sol (hunk 2 above)
Comment (AugustoL) [reply contracts/LifCrowdsale.sol]: :+1:

File: contracts/LifCrowdsale.sol (hunk 3 above)
Comment (AugustoL) [reply contracts/LifCrowdsale.sol]: we can add one, yes. It will be easier.

**Pull request merged**